src/raygun.rum/index.js

```diff
@@ -958,6 +984,81 @@ var raygunRumFactory = function(window, $, Raygun) {
       makePostCorsRequest(url, stringifiedPayload, successCallback, errorCallback);
     }
 
+     function updateUserAgentData(payload) {
+       if (!payload.eventData) { return; }
+
+       for (var i = 0; i < payload.eventData.length; i++) {
+         if (!!payload.eventData[i].data && Array.isArray(payload.eventData[i].data)) {
+
+           if (!!payload.eventData[i].device) {
+             payload.eventData[i].device = getHighFidelityUAString(payload.eventData[i].device);
+           }
+
+           for (var k = 0; k < payload.eventData[i].data.length; k++) {
+             var dataFragment = payload.eventData[i].data[k];
+
+             if (!!dataFragment.userAgent) {
+              dataFragment.userAgent = getHighFidelityUAString(dataFragment.userAgent);
+             }
+
+             if (!!window.raygunUserAgentData && !!window.raygunUserAgentData.platformVersion) {
+               var platformVersion = window.raygunUserAgentData.platformVersion.split(".");
+               dataFragment.device = {
+                 Family: window.raygunUserAgentData.platform,
+                 Major: platformVersion[0] || '',
+                 Minor: platformVersion[1] || '',
+                 Patch: platformVersion[2] || '',
+                 PatchMinor: platformVersion[3] || ''
+               };
+             }
+           }
+         }
+       }
+     }
+
+    function getHighFidelityUAString(userAgentString) {
+
+      if (!window.raygunUserAgentData) {
+        return userAgentString;
+      }
+      if (window.raygunUserAgentData.platform === "Windows") {
+        var platformVersion = window.raygunUserAgentData.platformVersion.split(".");
+        if (parseInt(platformVersion[0] || '0') >= 13) {
```

Comment:
Can we pass the radix to `parseInt`? MDN has a good section on all the things that can go wrong when omitting the radix: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt#description

So instead of `parseInt(platformVersion[0] || '0')`, we would do `parseInt(platformVersion[0] || '0', 10)`

Reply:
Good catch thanks!